### PR TITLE
Improve banner and redirection logic for Sourcegraph App

### PR DESCRIPTION
**Pull request by vovakulikov**
Based on https://github.com/sourcegraph/sourcegraph/pull/49367

Prior to this PR in Sourcegraph App, if you run an instance with some local repositories, there was no notification about adding remote repositories (global nav bar notification) that would lead to the setup wizard. This PR adds this notification in case if you have no remote repositories 

## Test plan
- Run fresh Sourcegraph instance (not app instance)
- You should see the common flow be default (wizard is required unless you click to skip the wizard button)
- Run fresh Sourcegraph instance 
- If you run this and there are already some local repositories, you should see the home page but with a global setup banner with a link to the setup wizard
- After you add some remote repositories, you should see no banner 

## App preview:

- [Web](https://sg-web-vk-fix-app-setup-banner.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.14 kb) | 0.00% (+0.14 kb) | -0.00% (-0.01 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/fix-app-setup-banner/compare-report.html;tab=live_object) for a full comparison between the commits 5ef3d7f and adede8d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by gl-srgr** [comment]
I merged my PR https://github.com/sourcegraph/sourcegraph/pull/49367 so the base branch changed. Not sure if you are able to reconcile any conflicts with the additions to Files Changed

File: client/web/src/integration/jscontext.ts
Comment (st0nebraker) [inline client/web/src/integration/jscontext.ts:1]: Is `context.needsRepositoryConfiguration` used anymore? Should we remove it from this file & `client/web/src/jscontext.ts` ?

**Review by st0nebraker — APPROVED** [review]
Looks good! I'm seeing expected behavior. Just a question below about `window.context.needsRepositoryConfiguration` deprecation

**Pull request merged**